src/components/LessonGroups/index.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import {map} from 'lodash'
+import {
+  newLessonsActionText,
+  inProgressTitleText,
+  inReviewTitleText,
+  inQueueTitleText,
+  finishedTitleText,
+  inProgressDescriptionText,
+  inReviewDescriptionText,
+  selfReviewDescriptionText,
+  inQueueDescriptionText,
+  finishedDescriptionText,
+  noInProgressLessonsDescriptionText,
+  noInReviewLessonsDescriptionText,
+  noInQueueLessonsDescriptionText,
+  noFinishedLessonsDescriptionText,
+} from 'utils/text'
+import {hasUnlockedSelfReview} from 'utils/milestones'
+import LessonList from 'components/LessonList'
+import Tabs from 'components/Tabs'
+import Prompt from 'components/Prompt'
+
+export default ({instructor}) => {
+
+  const items = [
+    {
+      title: inProgressTitleText,
+      description: inProgressDescriptionText,
+      states: [
+        'accepted',
+        'claimed',
+        'rejected',
+      ],
+      fallbackDescription: noInProgressLessonsDescriptionText,
+    },
+    {
+      title: inReviewTitleText,
+      description: inReviewDescriptionText
+        + (instructor && hasUnlockedSelfReview(instructor.published_lessons)
+          ? ` ${selfReviewDescriptionText}`
+          : ''),
+      states: [
+        'proposed',
+        'submitted',
+        'updated',
+      ],
+      fallbackDescription: noInReviewLessonsDescriptionText,
+    },
+    {
+      title: inQueueTitleText,
+      description: inQueueDescriptionText,
+      states: [
+        'approved'
+      ],
+      fallbackDescription: noInQueueLessonsDescriptionText,
+    },
+    {
+      title: finishedTitleText,
+      description: finishedDescriptionText,
+      states: [
+        'published',
+        'flagged',
+        'revised',
+      ],
+      fallbackDescription: noFinishedLessonsDescriptionText,
+    },
+  ]
```

Comment:
Broke up all the lesson states into logical groups for the tabs. The only ones missing are `requested` which is used in the `RequestedLessons` component separately, and the unused lesson states (`cancelled` and `retired`).